certz/certz.proto

```diff
@@ -185,6 +220,30 @@ service Certz {
   rpc Rotate(stream RotateCertificateRequest)
       returns (stream RotateCertificateResponse);
 
+  // AddSSLProfile is part of SSL profile management and allows for adding new
+  // SSL profile.
+  // When a SSL profile is added all its elements, i.e. certificat, CA trust
+  // bundle and a set of certificate revocation lists are created and they are
+  // empty, so before they can be used they have to be 'rotated' using
+  // the `Rotate()` RPC.
+  // Note that an attempt to add an already exisitng profile will be rejected
+  // with an error.
+  rpc AddSSLProfile(AddSSLProfileRequest) returns (AddSSLProfileResponse);
+
+  // DeleteSSLProfile is part of SSL profile management and allows for removing
```

Comment:
does should delete be failed of any service currently has that profile active?
Seems like the other behavior is annoying to deal with in likely you could delete the profile which then leaves in you a race condition of - should this force a reload of the servers which have the profile or do they go into lame duck / handle remaining rpc's then reset?

since that profile no longer exists .. then does the sever go into a failed state? or?

but if we just fail the delete request with "in use" it is very simple

Reply:
Yeah. That is a very good point but it is a bit in conflict with the fact that those profiles are assigned via gNMI. I can add a paragraph describing that the setting of SSL profile IDs have to be coordinated with gNSI but is this enough?